From efc11c4ac50ac51cd71d4dcfa84e63b11000c3f5 Mon Sep 17 00:00:00 2001
From: Ian McInerney <ian.s.mcinerney@ieee.org>
Date: Fri, 8 Apr 2022 01:08:12 +0100
Subject: [PATCH] Remove custom languages in wx 3.1.6 and above

Basque was added as a language with identifier "eu" into wx 3.1.6, so
there is no more need to provide custom languages. Additionally, this
struct changed formatting in 3.1.6, meaning it threw a compile error
when included (so it would have to be fixed if still needed).
---
 src/AudacityApp.cpp | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/AudacityApp.cpp b/src/AudacityApp.cpp
index d6e2576bf48..e3979249ceb 100644
--- a/src/AudacityApp.cpp
+++ b/src/AudacityApp.cpp
@@ -992,12 +992,10 @@ void AudacityApp::OnTimer(wxTimerEvent& WXUNUSED(event))
 #define WL(lang,sublang)
 #endif
 
-#if wxCHECK_VERSION(3, 0, 1)
+#if wxCHECK_VERSION(3, 0, 1) && !wxCHECK_VERSION(3, 1, 6)
 wxLanguageInfo userLangs[] =
 {
-   // Bosnian is defined in wxWidgets already
-//   { wxLANGUAGE_USER_DEFINED, wxT("bs"), WL(0, SUBLANG_DEFAULT) wxT("Bosnian"), wxLayout_LeftToRight },
-
+   // Included upstream in version 3.1.6
    { wxLANGUAGE_USER_DEFINED, wxT("eu"), WL(0, SUBLANG_DEFAULT) wxT("Basque"), wxLayout_LeftToRight },
 };
 #endif
@@ -1332,7 +1330,7 @@ bool AudacityApp::OnInit()
    //
    // TODO:  The whole Language initialization really need to be reworked.
    //        It's all over the place.
-#if wxCHECK_VERSION(3, 0, 1)
+#if wxCHECK_VERSION(3, 0, 1) && !wxCHECK_VERSION(3, 1, 6)
    for (size_t i = 0, cnt = WXSIZEOF(userLangs); i < cnt; i++)
    {
       wxLocale::AddLanguage(userLangs[i]);
